### PR TITLE
split some document-related code from core.js to src/Document.js

### PR DESCRIPTION
**Pull request by ngokevin**
Continuing some of the split. Later, could move Window to its own module.

### CODE REVIEW

**Comment by ngokevin** [comment]
Rebased.

File: src/Document.js [hunk 1]
```diff
@@ -1,15 +1,199 @@
...
+        return new MouseEvent();
+      case 'Event':
+      case 'Events':
+      case 'HTMLEvents':
+        return new Event();
+      default:
+        throw new Error('invalid createEvent type: ' + type);
+    }
+  };
+  document.createRange
```
Comment (modulesio) [inline src/Document.js]: If the point is a TODO, can we comment that?

File: src/Document.js [hunk 2]
```diff
@@ -1,15 +1,199 @@
...
+  const ast = parse5.parse(s, {
+    sourceCodeLocationInfo: true,
+  });
+  ast.tagName = 'document';
+  return _parseDocumentAst(ast, options, window, true);
+};
+module.exports._parseDocument = _parseDocument;
+GlobalContext._parseDocument = _parseDocument;
+
+const _parseDocumentAst = (ast, options, window, uppercase) => {
```
Comment (modulesio) [inline src/Document.js]: I guess that means the options arg is dead.

File: src/Document.js (hunk 1 above)
Comment (ngokevin) [reply src/Document.js]: Oops, accidentally got in there...

File: src/Document.js (hunk 2 above)
Comment (ngokevin) [reply src/Document.js]: We can pass it down later if it ends up getting used?

Comment (modulesio) [reply src/Document.js]: I think it's attached to the `window`, which also comes in here.

I'll clean it up; not really related to this PR. 👍

**Pull request merged**